### PR TITLE
gh-81094: Refer to PEP 318 in compound_statements.rst

### PR DESCRIPTION
**Pull request by Rasputin2**
…er Function definitions section.

----
📚 Documentation preview 📚: https://cpython-previews--113588.org.readthedocs.build/en/113588/reference/compound_stmts.html

* Issue: gh-81094

### CODE REVIEW

**Review by hugovk — APPROVED** [review]
Thanks, looks good!

Let's give a bit of time before merge in case @slateny want to check it too.

File: Doc/reference/compound_stmts.rst
Comment (slateny) [inline Doc/reference/compound_stmts.rst:1372]: LGTM as well, thanks. Optionally, it might be useful to mirror the "Class definitions" section and also reference PEP 3129 under the "Function definitions" block with something like

```suggestion
      Function and method decorators were introduced. Class decorators were introduced in :pep:`3129`.
```

I'll let Hugo decide whether that addition's useful or not 🙂 

Since we're changing this section anyways, on line 1365 there's a missing period after `variables and instance variables` that my suggestion can't quite reach. If you could give that a quick fix, that'd be great too!

**Comment by Rasputin2** [comment]
I just did another pull request to fix the sentence.  Let me know if you want me to do that other change. I am working tomorrow anyway.
I am eager to contribute, as I work with Machine Learning / AI stuff and business process automation, and so I have ideas about more advanced stuff to contribute (i.e., to things like openpyxl and python-pptx and natural language processing), but I realize I have to start with the easy stuff and then work my way up.
John D. McDonald
PJ Consulting LLCPh.      1-312-945-3518Mobile 1-312-479-2854 

    On Saturday, December 30, 2023 at 10:49:35 PM CST, Stanley ***@***.***> wrote:  
 
 


@slateny commented on this pull request.

In Doc/reference/compound_stmts.rst:
> @@ -1368,6 +1368,8 @@ access the local variables of the function containing the def.  See section
       Support for forward references within annotations by preserving
       annotations in a string form at runtime instead of eager evaluation.
 
+   :pep:`318` - Decorators for Functions and Methods
+      Function and method decorators were introduced.

LGTM as well, thanks. Optionally, it might be useful to mirror the "Class definitions" section and also reference PEP 3129 under the "Function definitions" block with something like
⬇️ Suggested change-      Function and method decorators were introduced.
+      Function and method decorators were introduced. Class decorators were introduced in :pep:`3129`.

I'll let Hugo decide whether that addition's useful or not 🙂

Since we're changing this section anyways, on line 1365 there's a missing period after variables and instance variables that my suggestion can't quite reach. If you could give that a quick fix, that'd be great too!

File: Doc/reference/compound_stmts.rst
Comment (hugovk) [reply Doc/reference/compound_stmts.rst:1372]: Sure, let's include this.

@Rasputin2 Please could you add that? You can click the "Commit suggestion" button to apply this.

And add the period as a new commit in this PR, rather than a new one.

Thanks!

**Comment by Rasputin2** [comment]
Ok, I think (hope) I did this correctly.  See below.  Another reason for volunteering is I am hoping to improve my Github skills.  Like I didn't even know you could do more than one commit per pull request.



John D. McDonald
PJ Consulting LLCPh.      1-312-945-3518Mobile 1-312-479-2854 

    On Sunday, December 31, 2023 at 09:21:40 AM CST, Hugo van Kemenade ***@***.***> wrote:  
 
 


@hugovk commented on this pull request.

In Doc/reference/compound_stmts.rst:
> @@ -1368,6 +1368,8 @@ access the local variables of the function containing the def.  See section
       Support for forward references within annotations by preserving
       annotations in a string form at runtime instead of eager evaluation.
 
+   :pep:`318` - Decorators for Functions and Methods
+      Function and method decorators were introduced.

Sure, let's include this.

@Rasputin2 Please could you add that? You can click the "Commit suggestion" button to apply this.

And add the period as a new commit in this PR, rather than a new one.

Thanks!

File: Doc/reference/compound_stmts.rst
Comment (hugovk) [inline Doc/reference/compound_stmts.rst:1372]: Please add this bit:
```suggestion
      Function and method decorators were introduced.
      Class decorators were introduced in :pep:`3129`.
```

**Comment by Rasputin2** [comment]
I believe I accepted these changes as part of the same PR.  So, I think this is done, unless you guys disagree.
John D. McDonald
PJ Consulting LLCPh.      1-312-945-3518Mobile 1-312-479-2854 

    On Monday, January 1, 2024 at 03:08:05 AM CST, Hugo van Kemenade ***@***.***> wrote:  
 
 


@hugovk commented on this pull request.

In Doc/reference/compound_stmts.rst:
>  
    :pep:`563` - Postponed Evaluation of Annotations
       Support for forward references within annotations by preserving
       annotations in a string form at runtime instead of eager evaluation.
 
+   :pep:`318` - Decorators for Functions and Methods
+      Function and method decorators were introduced.

Please add this bit:
⬇️ Suggested change-      Function and method decorators were introduced.
+      Function and method decorators were introduced.
+      Class decorators were introduced in :pep:`3129`.

**Review by hugovk — APPROVED** [review]
Thank you!

**Pull request merged**
